ast.c

```diff
@@ -1381,6 +1396,64 @@ PHP_METHOD(ast_Node, __construct) {
 	}
 }
 
+PHP_METHOD(ast_Node, parseCode) {
+	zend_string *code, *filename = NULL;
+	zend_long version = -1;
+	zend_class_entry *node_class;
+
+	ZEND_PARSE_PARAMETERS_START(2, 3)
+		Z_PARAM_STR(code)
+		Z_PARAM_LONG(version)
+		Z_PARAM_OPTIONAL
+		Z_PARAM_STR(filename)
+	ZEND_PARSE_PARAMETERS_END();
+
+	if (ast_check_version(version) == FAILURE) {
+		return;
+	}
+
+	node_class = zend_get_called_scope(execute_data);
```

Comment:
This copied nearly verbatim from https://github.com/php/php-src/blob/59d257d1aea70e370da5937cc019ca5bdcbb8d46/ext/tokenizer/tokenizer.c#L103